### PR TITLE
feat: Add `serialize_compute` and `unserialize_compute` to `BaseBone`

### PR DESCRIPTION
**Pull request by ArneGudermann**
fix for #1138

### CODE REVIEW

File: src/viur/core/bones/base.py [hunk 1]
```diff
@@ -707,6 +685,29 @@ def serialize(self, skel: 'SkeletonInstance', name: str, parentIndexed: bool) ->
             return True
         return False
 
+    def serialize_compute(self, skel, name):
```
Comment (sveneberth) [inline src/viur/core/bones/base.py:688]:
```suggestion
    def serialize_compute(self, skel: "SkeletonInstance", name: str) -> None:
```

File: src/viur/core/bones/base.py [hunk 2]
```diff
@@ -849,6 +819,37 @@ def unserialize(self, skel: 'viur.core.skeleton.SkeletonInstance', name: str) ->
         skel.accessedValues[name] = res
         return True
 
+    def unserialize_compute(self, skel, name, loaded_value):
```
Comment (sveneberth) [inline src/viur/core/bones/base.py:822]:
```suggestion
    def unserialize_compute(self, skel: "SkeletonInstance", name: str, loaded_value: t.Any) -> None:
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: src/viur/core/bones/base.py [hunk 3]
```diff
@@ -849,6 +819,37 @@ def unserialize(self, skel: 'viur.core.skeleton.SkeletonInstance', name: str) ->
         skel.accessedValues[name] = res
         return True
 
+    def unserialize_compute(self, skel: "SkeletonInstance", name: str, loaded_value: t.Any) -> None:
```
Comment (sveneberth) [inline src/viur/core/bones/base.py:822]:
```suggestion
    def unserialize_compute(self, skel: "SkeletonInstance", name: str, loaded_value: t.Any) -> bool:
```

**Review by sveneberth — DISMISSED** [review]

**Review by phorward — CHANGES_REQUESTED** [review]
@ArneGudermann can you please fix the conflicts with the current develop branch, so we can merge this?

File: src/viur/core/bones/base.py [hunk 4]
```diff
@@ -781,6 +759,29 @@ def serialize(self, skel: 'SkeletonInstance', name: str, parentIndexed: bool) ->
             return True
         return False
 
+    def serialize_compute(self, skel: "SkeletonInstance", name: str) -> None:
+        # Handle compute on write
```
Comment (phorward) [inline src/viur/core/bones/base.py:763]: Can you provide an adequate doc-string here?

File: src/viur/core/bones/base.py [hunk 5]
```diff
@@ -935,6 +895,52 @@ def transact():
         skel.accessedValues[name] = res
         return True
 
+    def unserialize_compute(self, skel: "SkeletonInstance", name: str, loaded_value: t.Any) -> bool:
+        # Is this value computed?
```
Comment (phorward) [inline src/viur/core/bones/base.py:899]: Can you provide an adequate doc-string here?

**Review by phorward — CHANGES_REQUESTED** [review]

File: src/viur/core/bones/base.py [hunk 6]
```diff
@@ -965,6 +932,58 @@ def transact():
...
+        match self.compute.interval.method:
+            # Computation is bound to a lifetime?
+            case ComputeMethod.Lifetime:
+                now = utils.utcNow()
+
+                # check if lifetime exceeded
+                last_update = skel.dbEntity.get(f"_viur_compute_{name}_")
+                skel.accessedValues[f"_viur_compute_{name}_"] = last_update or now
+
+
```
Comment (phorward) [inline src/viur/core/bones/base.py:957]:
```suggestion
```

File: src/viur/core/bones/base.py [hunk 7]
```diff
@@ -790,26 +790,33 @@ def serialize(self, skel: 'SkeletonInstance', name: str, parentIndexed: bool) ->
...
-                case ComputeMethod.OnWrite:
-                    skel.accessedValues[name] = self._compute(skel, name)
+        """
+        This function checks whether a bone is computed and if this is the case, it attempts to serialize the
+        value with the appropriate calculation method
 
-                case ComputeMethod.Lifetime:
-                    now = utils.utcNow()
+        :param skel : The SkeletonInstance where the current Bone is located
+        :param name: The name of the Bone in the Skeleton
```
Comment (sveneberth) [inline src/viur/core/bones/base.py:798]:
```suggestion
        :param skel: The SkeletonInstance where the current bone is located
        :param name: The name of the bone in the Skeleton
```

File: src/viur/core/bones/base.py [hunk 8]
```diff
@@ -924,34 +933,54 @@ def unserialize(self, skel: 'viur.core.skeleton.SkeletonInstance', name: str) ->
...
+        """
+        This function checks whether a bone is computed and if this is the case, it attempts to deserialise the
+        value with the appropriate calculation method
+
+        :param skel : The SkeletonInstance where the current Bone is located
+        :param name: The name of the Bone in the Skeleton
+        :param loaded_value: The value from the DB Entity
+        :return: True if the Bone was unserialized, False otherwise
+        """
+        if not self.compute and self._prevent_compute:
```
Comment (sveneberth) [inline src/viur/core/bones/base.py:945]: The negation of
```py
self.compute and not self._prevent_compute
```


# (
**=**
```py
not (self.compute and not self._prevent_compute)
```
# )
is after de Morgan's laws
```suggestion
        if not self.compute or self._prevent_compute:
```

**Review by sveneberth — DISMISSED** [review]

**Review by phorward — APPROVED** [review]

File: src/viur/core/bones/base.py [hunk 9]
```diff
@@ -881,43 +889,9 @@ def unserialize(self, skel: 'viur.core.skeleton.SkeletonInstance', name: str) ->
...
-                        else:
-                            db.RunInTransaction(transact)
-
-                        return True
-
-                # Compute on every deserialization
-                case ComputeMethod.Always:
-                    skel.accessedValues[name] = self._compute(skel, name)
-                    return True
+
```
Comment (ArneGudermann) [inline src/viur/core/bones/base.py:892]:
```suggestion
```

**Pull request merged**